### ISSUE TITLE
Correctly escape back slashes in single quoted strings

### ISSUE THREAD

**Issue by leofeyer** [contao/core-bundle#372]
We are often doing the following:

```php
private $mapper = [
    'Contao\\CoreBundle\\Exception\\AccessDeniedException' => 'AccessDeniedHttpException',
    'Contao\\CoreBundle\\Exception\\ForwardPageNotFoundException' => 'InternalServerErrorHttpException',
    'Contao\\CoreBundle\\Exception\\IncompleteInstallationException' => 'InternalServerErrorHttpException',
];
```

But in fact, the double back slashes are wrong here. According to [the PHP manual](http://php.net/manual/en/language.types.string.php#language.types.string.syntax.single), the back slash only needs to be escaped if it is followed by a single quote:

```php
$a = 'No need to escape the \ here.';
$b = 'The following character needs to be escaped here: \\';
```

We should fix this in the non-legacy code.

**Comment by aschempp** [comment]
The double backslash is not wrong, it just also works *most of the time* with a single backslash.

You’re not the only person with that question:
http://stackoverflow.com/questions/25459197/single-or-double-slash-when-using-php-namespace

Double backslashes is just what everyone uses, probably similar to Yoda conditions. It’s especially important for double quoted strings, as "test\namespace" would actually produce a newline character (\n).

**Comment by leofeyer** [comment]
I am talking about single quoted strings though.

**Comment by aschempp** [comment]
I know, but it’s just best practice imho. I remember I always did it „wrong“ and got corrected by @Toflar

**Comment by leofeyer** [comment]
I'll change it away :)

**Comment by leofeyer** [comment]
It is the same with the regular expressions. You can escape all kind of characters but it is a lot more readable if you only escape what really needs to be escaped (see contao/core@fd601faf7ed864537c92d8e07a46b776a6f2a91a).

**Comment by discordier** [comment]
I'd keep the backslashes. I always copy them around from PHP to i.e. composer.json and having two of them works in both strings where as having only a single one does not.

**Comment by leofeyer** [comment]
I have done a search for `Symfony\` in string literals in PhpStorm to see which variant they are using. The vast majority (> 90%) does not use double backslashes, which is correct IMHO.

You are right that Composer and the PHP `var_export()` function are using double backslashes, but this is just because it is way easier to apply `addcslashes()` to an unknown input string than to specifically quote only the backslash right in front of the closing single quote.

**Comment by leofeyer** [comment]
Fixed in 79ee484ce859beef0c06ff1832e84dcab3641479.
